zcash_client_sqlite/src/wallet.rs

```diff
@@ -1298,45 +1358,18 @@ pub(crate) fn get_unspent_transparent_outputs<P: consensus::Parameters>(
     let addr_str = address.encode(params);
 
     let mut utxos = Vec::<WalletTransparentOutput>::new();
-    let mut rows = stmt_blocks.query(named_params![
+    let mut rows = stmt_utxos.query(named_params![
         ":address": addr_str,
         ":max_height": u32::from(max_height),
         ":stable_height": u32::from(stable_height),
     ])?;
     let excluded: BTreeSet<OutPoint> = exclude.iter().cloned().collect();
     while let Some(row) = rows.next()? {
-        let txid: Vec<u8> = row.get(0)?;
-        let mut txid_bytes = [0u8; 32];
-        txid_bytes.copy_from_slice(&txid);
-
-        let index: u32 = row.get(1)?;
-        let script_pubkey = Script(row.get(2)?);
-        let value_raw: i64 = row.get(3)?;
-        let value = NonNegativeAmount::from_nonnegative_i64(value_raw).map_err(|_| {
-            SqliteClientError::CorruptedData(format!("Negative utxo value: {}", value_raw))
-        })?;
-        let height: u32 = row.get(4)?;
-
-        let outpoint = OutPoint::new(txid_bytes, index);
-        if excluded.contains(&outpoint) {
+        let output = to_unspent_transparent_output(row)?;
+        if excluded.contains(output.outpoint()) {
             continue;
         }
```

Comment:
There must be a more functional way to express this, probably something like:
```rust
let utxos: Vec<_> = rows.filter_map(|row| {
    let output = to_unspent_transparent_output(row)?;
    if excluded.contains(output.outpoint()) {
        None
    } else {
        Some(output)
    }
}).collect();
```

Reply:
`Rows` isn't a proper iterator. Unfortunately this is probably the best way, given the restrictions of rusqlite.